contributing/new-bindings.rst

```diff
@@ -0,0 +1,89 @@
+Adding support for other programming languages
+==============================================
+
+Ambition
+--------
+
+We'd like Xapian to be usable from as many languages as possible. However,
+this means more than just having bindings code. To truly support a language,
+we need to have:
+
+* working bindings that can be built against a latest version of Xapian and the
+  latest version of the target language
+
+* documentation to help people familiar with the target language to start using
+  the bindings
+
+* example code -- ideally translations of the code in the `Xapian user guide`_.
+  You can start from the `user guide source code`_, which already supports a few
+  languages.
+
+* automated tests for the bindings; at least a "smoketest" hooked into the tests
+  for the build system, so we'll notice
```

Comment:
Unfinished paragraph?

Reply:
One thing probably worth noting here is that for languages wrapped via SWIG we don't currently aim to test every single wrapped method, but rather rely on the xapian-core test suite to test xapian-core's behaviour via the C++ API, along with the typemap-based nature of SWIG's wrapping.  If we test one method which takes a `const std::string&` parameter then that provides test coverage for the typemap which handles that and we gain very little over that by testing every method which takes `const std::string&`.

So it's particularly important to have coverage for custom typemaps, and for anything else where we customise what SWIG produces by default.  It's also good to cover the SWIG built-in typemaps, so we catch any changes there which affect us.

And we also want regression tests for any bugs fixed.